File: Bitwise XOR of All Pairings.py
class Solution:
    def xorAllNums(self, nums1: List[int], nums2: List[int]) -> int:
        n1 = len(nums1)
        n2 = len(nums2)
        ans = 0
        if n2 % 2 == 1:
            for num in nums1:
                ans ^= num
        if n1 % 2 == 1:
            for num in nums2:
                ans ^= num
        return ans

File: Bitwise XOR of All Pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int n1 = nums1.size(), n2 = nums2.size();
        int ans = 0;
        if (n1 % 2) {
            for (int num: nums2) {
                ans ^= num;
            }
        }

        if (n2 % 2) {
            for (int num: nums1) {
                ans ^= num;
            }
        }
        return ans;
    }
};
